### PR TITLE
Remove `parameterized_class` from SyntheticVolume tests

### PR DESCRIPTION
**Pull request by j-c-c**
This PR removes `parameterized_class` and unittest from `test_synthetic_volume.py`. Replaced with parametrized pytest fixtures. 

Related issue: #803

### CODE REVIEW

File: tests/test_synthetic_volume.py [hunk 1]
```diff
@@ -14,127 +14,120 @@
...
-class AsymmetricVolumeCase(Base, TestCase):
-    vol_class = AsymmetricVolume
-    L = 20
-
-
-@parameterized_class(("L"), [(21,)])
-class LegacyVolumeCase(Base, TestCase):
-    vol_class = LegacyVolume
-    L = 20
+        assert abs(corr - 1) < 1e-5
```
Comment (garrettwrong) [inline tests/test_synthetic_volume.py:137]: Lets either make a debug log or a string after the assert with `abs(corr -1 )`, so if it fails we can get an idea by how much without doing any additional work.

File: tests/test_synthetic_volume.py [hunk 2]
```diff
@@ -14,127 +14,120 @@
...
+    vol_kwargs = dict(
+        L=res,
+        C=1,
+        seed=0,
+        dtype=dtype_fixture,
+    )
+    if len(params) > 2:
+        vol_kwargs["order"] = params[2]
+
+    return vol_class(**vol_kwargs)
```
Comment (garrettwrong) [inline tests/test_synthetic_volume.py:88]: that worked out pretty clean, nice.

File: tests/test_synthetic_volume.py [hunk 3]
```diff
@@ -14,127 +14,120 @@
...
+
+
+# SyntheticVolume tests:
+def test_volume_repr(vol_fixture):
+    """Test Synthetic Volume repr"""
+    assert repr(vol_fixture).startswith(f"{vol_fixture.__class__.__name__}")
```
Comment (garrettwrong) [inline tests/test_synthetic_volume.py:94]: lets debug log it too

File: tests/test_synthetic_volume.py [hunk 4]
```diff
@@ -14,127 +14,120 @@
...
+    (DnSymmetricVolume, 21, 2),
+    (DnSymmetricVolume, 40, 3),
+    (DnSymmetricVolume, 41, 3),
+    (DnSymmetricVolume, 42, 4),
+    (DnSymmetricVolume, 43, 4),
+    (DnSymmetricVolume, 55, 5),
+    (DnSymmetricVolume, 56, 5),
+    (DnSymmetricVolume, 64, 6),
+    (DnSymmetricVolume, 65, 6),
+]
```
Comment (garrettwrong) [inline tests/test_synthetic_volume.py:53]: Note that you could code these out using `resolution%10`, but I wouldn't bother, would end up being similar amount of lines by the time you add two additional lists.

**Review by garrettwrong — CHANGES_REQUESTED** [review]
This looks good, thanks!  Just two minor tweaks and two comments.

File: tests/test_synthetic_volume.py (hunk 1 above)
Comment (j-c-c) [reply tests/test_synthetic_volume.py:137]: This value is reported to us in the test summary in the event of a fail (see below). This is one advantage of using pytest over unittest.

```
    def test_volume_symmetry(vol_fixture):
        """Test that volumes have intended symmetry."""
        vol = vol_fixture.generate()
    
        # Rotations in symmetry group, excluding the Identity.
        rots = vol_fixture.symmetry_group.matrices[1:]
    
        for rot in rots:
            # Rotate volume by an element of the symmetric group.
            rot_vol = vol.rotate(Rotation(rot), zero_nyquist=False)
    
            # Check that correlation is close to 1.
            corr = np.dot(rot_vol[0].flatten(), vol[0].flatten()) / np.dot(
                vol[0].flatten(), vol[0].flatten()
            )
>           assert abs(corr - 1) < 1e-7
E           assert 1.7881393432617188e-07 < 1e-07
E            +  where 1.7881393432617188e-07 = abs((0.9999998 - 1))

```

File: tests/test_synthetic_volume.py (hunk 2 above)
Comment (j-c-c) [reply tests/test_synthetic_volume.py:88]: thanks!

File: tests/test_synthetic_volume.py (hunk 4 above)
Comment (j-c-c) [reply tests/test_synthetic_volume.py:53]: noted!

File: tests/test_synthetic_volume.py (hunk 1 above)
Comment (garrettwrong) [reply tests/test_synthetic_volume.py:137]: right! I keep forgetting... I got so used to unittest while I was on this project. Thanks!

File: tests/test_synthetic_volume.py (hunk 3 above)
Comment (j-c-c) [reply tests/test_synthetic_volume.py:94]: Same as above, we also get a detailed message on the failed assert:

```
_______________________________________ test_volume_repr[LegacyVolume, res=21-float32] _______________________________________

vol_fixture = LegacyVolume {'order': 1, 'K': 16, 'alpha': 1.0, 'L': 21, 'C': 1, 'seed': 0, 'dtype': <class 'numpy.float32'>}

    def test_volume_repr(vol_fixture):
        """Test Synthetic Volume repr"""
>       assert repr(vol_fixture).startswith(f"bad stuff {vol_fixture.__class__.__name__}")
E       assert False
E        +  where False = <built-in method startswith of str object at 0x7f7c4df68850>('bad stuff LegacyVolume')
E        +    where <built-in method startswith of str object at 0x7f7c4df68850> = "LegacyVolume {'order': 1, 'K': 16, 'alpha': 1.0, 'L': 21, 'C': 1, 'seed': 0, 'dtype': <class 'numpy.float32'>}".startswith
E        +      where "LegacyVolume {'order': 1, 'K': 16, 'alpha': 1.0, 'L': 21, 'C': 1, 'seed': 0, 'dtype': <class 'numpy.float32'>}" = repr(LegacyVolume {'order': 1, 'K': 16, 'alpha': 1.0, 'L': 21, 'C': 1, 'seed': 0, 'dtype': <class 'numpy.float32'>})

tests/test_synthetic_volume.py:94: AssertionError
```

Comment (garrettwrong) [reply tests/test_synthetic_volume.py:94]: When it passes, do you know what the repr is from the logs?

Comment (j-c-c) [reply tests/test_synthetic_volume.py:94]: Nope. I've added a debug log here.

Comment (garrettwrong) [reply tests/test_synthetic_volume.py:94]: Thanks :)

**Review by garrettwrong — APPROVED** [review]
Thanks. Another batch of `parameterized` gone :).

**Pull request merged**